tic-tac-toe/src/urbit/app/toe.hoon

```diff
@@ -0,0 +1,1176 @@
+          ::  $who: player that can perform a move
+          ::
+          who=ship
+          ::  $next: flag to indicate a replay
+          ::
+          next=?
+          ::
+          ::  $front: id for frontend connection
+          ::
+          front=bone
+          ::  $consol: console state
+          ::
+          ::     $conn:  id for console connection
+          ::     $state: what's in the console
+          ::
+          consol=[conn=bone state=sole-share]
+      ==
+    ::
+    +$  move  (pair bone card)
+    ::
+    +$  card
+      $%  [%diff diff-data]
+          [%peer wire dock path]
+          [%pull wire dock ~]
+          [%poke wire dock poke-data]
+          [%wait wire p=@da]
+          [%http-response =http-event:http]
+          [%connect wire binding:eyre term]
+      ==
+    ::
+    +$  diff-data
+      $%  ::  See %/mar/toe/* for specific details
+          ::    * = [turno, player, winner]
+          ::
+          ::    FIXME: replace wiht =toe-{mark} ?
+          ::
+          [%toe-turno toe-turno]
+          [%toe-player toe-player]
+          [%toe-winner toe-winner]
+          [%sole-effect sole-effect]
+          [%json json]
+       ==
+    ::
+    +$  poke-data
+      $%  [%toe-cancel toe-cancel]
+          [%launch-action [@tas path @t]]
+      ==
+    ::
+    ::  FIXME: $spot has to be redefined
+    ::     even though it's already in sur...
+    ::     it gave nest fails in +-moves-start
+    ::     ?:  (~(has by board.sat) (spot-val u.try))
+    ::     -find.a
+    ::     which i assume references a/(tree (pair)) from +by
+    ::
+    +$  spot  [coord coord]
+    --
+::
+::  %app
+::
+::    Our app is defined as a "door" (multi-armed core with a sample).
+::    Arms are grouped in chapters (+|) based on common functionality
+::
+|_  [bol=bowl:gall %0 sat=state]
+::
+::  %alias: rewording of commonly used nouns
+::
++|  %alias
+::
+::  %me: alias for our ship
+::
++*  me  our.bol
+::
+::  %ze: alias for the origin of an incoming request
+::
++*  ze  src.bol
+::
+::
+::  %this: common idiom to refer to our whole %app door and its context
+::
++*  this  .
+::
+::  %state
+::
+::    Arms to innitialize (and restart) our app's state
+::
++|  %state
+::
+++  prep
+  ::  TODO: we need to treat old state differently. how? good question
+  ::    rather than reseting old the time
+  ::
+  =>  |%
+      ++  states
+        $%  [%0 s=state]
+        ==
+      --
+  |=  old=(unit states)
+  ^-  (quip move _this)
+  =/  launcha=poke-data
+    [%launch-action [%toe /toetile '/~toe/js/tile.js']]
+  =/  moves=(list move)
+    :~  :: %connect here tells %eyre to mount at the /~toe endpoint.
+        ::
+        [ost.bol %connect / [~ /'~toe'] %toe]
+        [ost.bol %poke /toe [our.bol %launch] launcha]
+    ==
+  ?~  old
+    ::  we haven't modified the previous state
+    ::
+    (wipe moves)
+  ::  the old state needs to be adapted to the new one
+  ::
+  ?-  -.u.old
+    %0  (restore s.u.old moves)
```

Comment:
As your comment above hints at, this probably shouldn't be re-sending the setup moves.